route/bgmetadata.go

```diff
@@ -208,6 +212,58 @@ func (s *shard) saveShardState(cachePath string) error {
 	return nil
 }
 
+// Remove Shard State file from cache and return an error if file exists and but fails to be removed.
+func (s *shard) removeShardState(cachePath string) error {
+	fp := filepath.Join(cachePath, fmt.Sprintf("shard%d.state", s.num))
+
+	if _, err := os.Stat(fp); !os.IsNotExist(err) {
+		return os.Remove(fp)
+	}
+
+	return nil
+}
+
+// Continuously save bloom filters
+func (m *BgMetadata) saveBloomFilter() {
+	var err error
+
+	m.logger.Debug("Starting goroutine for bloom filter saving")
+	m.wg.Add(1)
+	defer m.wg.Done()
+
+	// Use an empty filter to check if existing filters are empty or not.
+	emptyFilter := bloom.NewWithEstimates(m.bfCfg.N, m.bfCfg.P)
```

Comment:
An empty filter has the  same size than a filter, with the settings we use the filter is about 4go, IMHO we don't need to  clear the filter, just overwrite it

Reply:
As we don't need anymore to know if the filter is empty to remove its saved state file, this part is now obsolete and was removed in 71caf3370d823b226ca4e592795c3d02c74b6463.